chore: vLLM Mock Streams + Parsing e2e tests

ayushag-nv opened the pull request:
#### Overview:

This PR 
- captures data streams for gpt-oss-20b, qwen3-0.6B and Nemotron 49b v1.5 required for tool calling and reasoning tests.
- Add test suite for vLLM streams

Stream Capture PR: https://github.com/ai-dynamo/dynamo/pull/3400

#### Details:

<!-- Describe the changes made in this PR. -->

#### Where should the reviewer start?

<!-- call out specific files that should be looked at closely -->

#### Related Issues: (use one of the action keywords Closes / Fixes / Resolves / Relates to)

- closes GitHub issue: 
- https://github.com/ai-dynamo/dynamo/issues/3396
- https://github.com/ai-dynamo/dynamo/issues/3398

ayushag-nv committed:
chore: added vllm tool + reason data

Signed-off-by: ayushag <ayushag@nvidia.com>

ayushag-nv commented:
/ok to test 7b18b5f60dd84bfcdc549ee59261a95b6b294ff3

grahamking reviewed (dismissed):
> +33,667 −0

We really don't want thirty three thousand lines of JSON. There must be a much smaller test case.

ayushag-nv committed:
chore: reduced lines v1

Signed-off-by: ayushag <ayushag@nvidia.com>

ayushag-nv committed:
chore: reduced lines v2

Signed-off-by: ayushag <ayushag@nvidia.com>

ayushag-nv committed:
chore: reduced lines v3

Signed-off-by: ayushag <ayushag@nvidia.com>

ayushag-nv commented:
@grahamking Did some cleaning 
- removed few files which were different content but representing same thing
- removed extra information like token info and stuff which is not needed. 
- reformatted file : each chunk into a single line for better review. 

The concept of capturing data like this is already there. Below file is an example : 
https://github.com/ai-dynamo/dynamo/blob/main/lib/llm/tests/data/replays/mistralai/mistral-7b-instruct-v0.3/completions/completion.streaming.1

Let me know if there are more concerns. Thanks.

ayushag-nv committed:
chore: added gpt-oss tests

Signed-off-by: ayushag <ayushag@nvidia.com>

ayushag-nv committed:
chore: file rename

Signed-off-by: ayushag <ayushag@nvidia.com>

ayushag-nv committed:
chore: complete test case suite

Signed-off-by: ayushag <ayushag@nvidia.com>

ayushag-nv commented:
/ok to test 000e2699df2ff0d43496388baa2e144196840419

elyasmnvidian reviewed (approved):
"request_id": "8f33c28b-cb52-4272-9ac5-0cb9f80386d3",
  "normal_content": "\n\n",
  "reasoning_content": "\nOkay, the user is asking for the weather in San Francisco in Celsius. Let me check the tools provided. There's a function called get_weather that requires location and unit. The location needs to be a city and state, which San Francisco, CA is. The unit they specified is Celsius. So I should call get_weather with location \"San Francisco, CA\" and unit \"celsius\". That should retrieve the current weather data in the requested format.\n",
  "tool_calls": [{"index": 0, "id": "call-61672e29-0cbf-45a0-b028-72389d0b2972", "type": "function", "function": {"name": "get_weather", "arguments": "{\"location\":\"San Francisco, CA\",\"unit\":\"celsius\"}"}}],
  "data": [
  
  
I think the JSON keys in the json files need to be renamed. We should rename "data", "tool_calls", etc.. or restructure it because it wasn't obvious to me that the "data" field was the input and the other fields were "expected_output". 

Maybe we structure it like:

"request_id":
"expected_output": { "normal_content": ... , "reasoning_content": ..., "tool_calls": ... }
"input_data": ...

thoughts?
